### PR TITLE
[merged] Add support for LABEL RUN_OPTS_FILE and ${RUN_OPTS}

### PR DESCRIPTION
**Pull request by adelton**
The main use is for containers that might want to determine some `docker run` option during `atomic install`, in their `install.sh`. For example, `/var/lib/${NAME}/docker-run-opts` can be populated with `--hostname=ipa.example.test`, making the subsequent `atomic run` invoke the `docker run` with container-specific option.

### CODE REVIEW

**Comment by rhatdan** [comment]
bot, add author to whitelist

File: Atomic/run.py [hunk 1]
```diff
@@ -38,6 +39,17 @@ def run(self):
...
                 args += self.command
+                opts_file = self._get_args("RUN_OPTS_FILE")
+                if opts_file:
+                    opts_file = self.sub_env_strings("".join(opts_file))
+                    if opts_file.startswith("/"):
+                        try:
+                            run_opts = open(opts_file, "r").read()
+                            os.environ["RUN_OPTS"] = run_opts
+                        except IOError:
+                            util.write_out("Failed to read RUN_OPTS_FILE %s" % opts_file)
```
Comment (rhatdan) [inline Atomic/run.py]: Raise ValueError with these.  We don't want to continue to run the container if this fails.

File: Atomic/run.py [hunk 2]
```diff
@@ -38,6 +39,17 @@ def run(self):
             args = self._get_args("RUN")
             if args:
                 args += self.command
+                opts_file = self._get_args("RUN_OPTS_FILE")
+                if opts_file:
+                    opts_file = self.sub_env_strings("".join(opts_file))
+                    if opts_file.startswith("/"):
+                        try:
+                            run_opts = open(opts_file, "r").read()
+                            os.environ["RUN_OPTS"] = run_opts
```
Comment (rhatdan) [inline Atomic/run.py]: I would prefer to not modify the current environ.  We are moving away from this in another pull request.  Could we add this to self.run_opts and set it in cmd_env?

Comment (rhatdan) [inline Atomic/run.py]: https://github.com/projectatomic/atomic/pull/521

File: Atomic/run.py [hunk 3]
```diff
@@ -38,6 +39,17 @@ def run(self):
...
+                        except IOError:
+                            util.write_out("Failed to read RUN_OPTS_FILE %s" % opts_file)
+                    else:
+                        util.write_out("Failed to read RUN_OPTS_FILE %s: not absolute path" % opts_file)
```
Comment (rhatdan) [inline Atomic/run.py]: Raise exception.

File: docs/atomic-run.1.md [hunk 4]
```diff
@@ -49,6 +49,9 @@ These defaults are suggested values for your container images.
 **SUDO_GID**
   The `SUDO_GID` environment variable.  This is useful with the docker `-u` option for user space tools.  If the environment variable is not available, the default GID of the value for `SUDO_UID` is used.  If this value is not available, the value of `/proc/self/loginuid` is used.
 
+**RUN_OPTS**
```
Comment (rhatdan) [inline docs/atomic-run.1.md]: How about we just call this OPTS_FILE and then we can use it for INSTALL, RUN, UNINSTALL. Any reason to have multiple?

File: Atomic/run.py (hunk 1 above)
Comment (adelton) [inline Atomic/run.py]: What if the file just does not exist -- do you want an error as well?

File: Atomic/run.py (hunk 2 above)
Comment (adelton) [inline Atomic/run.py]: Will try.

File: docs/atomic-run.1.md (hunk 4 above)
Comment (adelton) [inline docs/atomic-run.1.md]: OK, the LABEL can be OPTS_FILE. What should the "environment variable" in the `docker run` command instead of `${RUN_OPTS}` be called? `${OPTS}`? `${OPTS_FILE_CONTENT}`? Calling it just `${OPTS_FILE}` seems weird, as it won't be expanded to file name but to content of that file.

Comment (adelton) [inline docs/atomic-run.1.md]: > How about we just call this OPTS_FILE and then we can use it for INSTALL, RUN, UNINSTALL. Any reason to have multiple?

Truth be told, I am able to imagine just the `RUN` use case, when in `INSTALL` you set the file to be consumed in `RUN`. I'm a bit afraid that using one file for all three (`INSTALL`, `RUN`, and `UNINSTALL`) might cause clashes because `RUN` might need some other options than `UNINSTALL`.

Do you have some examples of uses of this mechanism for `INSTALL` and `UNINSTALL`?

Or do you envision the `LABEL`s being different for each phase, just the "variable" where it is exposed being the same (`OPTS_FILE`)?

File: Atomic/run.py (hunk 1 above)
Comment (rhatdan) [inline Atomic/run.py]: If you don't think this is an error then don't state anything.  If we want to report a problem it should be a write_err() anyways.

File: docs/atomic-run.1.md (hunk 4 above)
Comment (rhatdan) [inline docs/atomic-run.1.md]: I am just thinking that once we have it for one option people will want it for others.
Lets keep it as RUN_OPTS_FILE and ignore the others for now, and we will create them if needed.

File: Atomic/run.py (hunk 2 above)
Comment (adelton) [inline Atomic/run.py]: > #521

The https://github.com/projectatomic/atomic/pull/521 as of ac83a85115cc35f467b8c35802c3be0517ab1877 seems to break the tests.

Comment (adelton) [inline Atomic/run.py]: Bisecting the history, it was already 0cba782a03d110a154550fa5f1ad5a9e7cfacbd5 which introduced the regression.

**Comment by adelton** [comment]
Take fc775bbdd02ab326a9d994159b6fa1e488025239 addresses all comments except for the `os.environ` one. I will update this commit one https://github.com/projectatomic/atomic/pull/521 is merged.

**Comment by adelton** [comment]
Alternatively, if this PR looks otherwise fine, it could be merged and https://github.com/projectatomic/atomic/pull/521 could adapt the `os.environ["RUN_OPTS"]` changes in the future.

**Comment by rhatdan** [comment]
@rh-atomic-bot r+

You need to update the containerlablels standards patch

**Comment by adelton** [comment]
> You need to update the containerlablels standards patch

Where does that live?

**Comment by rhatdan** [comment]
@adelton https://github.com/projectatomic/ContainerApplicationGenericLabels

**Comment by adelton** [comment]
> @adelton https://github.com/projectatomic/ContainerApplicationGenericLabels

Thanks, file https://github.com/projectatomic/ContainerApplicationGenericLabels/pull/72.
